crates/fraud_proofs/global_merkle_root/storage/src/update.rs

```diff
@@ -305,6 +305,246 @@ impl TransactionOutputs for Transaction {
     }
 }
 
-// TODO(#2582): Add tests (https://github.com/FuelLabs/fuel-core/issues/2582)
-#[test]
-fn dummy() {}
+#[cfg(test)]
+#[allow(non_snake_case)]
+mod tests {
+    use super::*;
+
+    use fuel_core_storage::{
+        structured_storage::test::InMemoryStorage,
+        transactional::{
+            ReadTransaction,
+            WriteTransaction,
+        },
+        StorageAsRef,
+    };
+    use fuel_core_types::fuel_tx::{
+        Bytes32,
+        ContractId,
+        TxId,
+    };
+
+    use rand::{
+        rngs::StdRng,
+        Rng,
+        SeedableRng,
+    };
+
+    #[test]
+    fn process_output__should_insert_coin() {
+        let mut rng = StdRng::seed_from_u64(1337);
+
+        // Given
+        let mut storage: InMemoryStorage<Column> = InMemoryStorage::default();
+        let mut storage_tx = storage.write_transaction();
+        let mut update_tx = storage_tx.update_transaction();
```

Comment:
we have a terminology problem here.
We use the term transaction both for

The InMemoryStorageTransaction
The Fuel transactions in a block.
I feel that in the future this could create a lot of confusion in the code. This is particularly problematic for Transaction::Update (and maybe Transaction::Upgrade). 
I like how you used storage_tx here as it makes it clear. But `update_tx` got me a bit confused. Perhaps we can agree to a convension to prefix all "storage" transactions with `storage_`, and all fuel transactions with `fuel_`?
```suggestion
        let mut storage_tx = storage.write_transaction();
        let mut storate_update_tx = storage_tx.update_transaction();
```

Reply:
Yeah I am not too happy about this terminology collision either. I like the suggestion, let's use it.